src/tests/Microsoft.PowerFx.Interpreter.Tests/FormulaValueSerializerTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Drawing;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using Xunit;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    /* !JYL! How to serialize and test:
+     * ErroraValeu
+     * UntypedObjectValue
+     * OptionSetValue
+     */
+
+    public class FormulaValueSerializerTests : PowerFxTest
+    {
+        [Fact]
+        public async Task PrimitiveValueSerializerTest()
+        {
+            var dateTimeNow = DateTime.Parse("12/25/2022 12:34:56");
+            var newGuid = Guid.Parse("811bfd49-38f6-4fdd-9993-4ed7e7bef277");
+
+            var numberValue = FormulaValue.New(1);
+            var boolValue = FormulaValue.New(true);
+            var colorValue = new ColorValue(IRContext.NotInSource(FormulaType.Color), Color.Blue);
+            var dateTimeValue = FormulaValue.New(dateTimeNow);
+            var dateValue = FormulaValue.NewDateOnly(dateTimeNow.Date);
+            var guidValue = FormulaValue.New(newGuid);
+            var stringValue = FormulaValue.New("*SERIALIZER*");
+            var timeValue = FormulaValue.New(dateTimeNow.TimeOfDay);
+
+            Assert.Equal("1", numberValue.ToExpression());
+            Assert.Equal("true", boolValue.ToExpression());
+            Assert.Equal("\"*SERIALIZER*\"", stringValue.ToExpression());
+            Assert.Equal("DateTime(2022,12,25,12,34,56)", dateTimeValue.ToExpression());
+            Assert.Equal("Date(2022,12,25)", dateValue.ToExpression());
+            Assert.Equal("Time(12,34,56)", timeValue.ToExpression());
```

Comment:
I would make this test a [Theory] instead

Reply:
We changed the approach of how serialization is tested.